src/coreclr/vm/comdelegate.cpp

```diff
@@ -189,16 +189,57 @@ class ShuffleIterator
         if (m_currentByteStackIndex < m_argLocDesc->m_byteStackSize)
         {
             const unsigned byteIndex = m_argLocDesc->m_byteStackIndex + m_currentByteStackIndex;
-            index = byteIndex / TARGET_POINTER_SIZE;
+
+#if !defined(TARGET_OSX) || !defined(TARGET_ARM64)
             m_currentByteStackIndex += TARGET_POINTER_SIZE;
 
             // Delegates cannot handle overly large argument stacks due to shuffle entry encoding limitations.
+            index = byteIndex / TARGET_POINTER_SIZE;
             if (index >= ShuffleEntry::REGMASK)
             {
                 COMPlusThrow(kNotSupportedException);
             }
 
-            return -(int)byteIndex;
+            // Only Apple Silicon ABI currently supports unaligned stack argument shuffling
+            _ASSERTE(byteIndex == index * TARGET_POINTER_SIZE);
+            return index;
+#else
```

Comment:
Is there a doc reference for this logic?

Reply:
This is just a loop to emit stack load/store `ShuffleEntry`s.  Arm64 logic can write 1,2,4,or 8 bytes in a single instruction.

I'll add a comment.